### PR TITLE
@orta => Try to use Shared Web Credentials as early as possible.

### PR DESCRIPTION
**Pull request by alloy**
As seen in Slack.

### CODE REVIEW

**Comment by orta** [comment]
@katarina - have you looked through the videos at how these changes works?

From my perspective this is a much better on-boarding flow for people who have Artsy credentials saved already, and the same flow for people without.

File: Artsy_Tests/Extensions/UIViewController+Testing.m [hunk 1]
```diff
@@ -1,22 +1,9 @@
...
-@property (nonatomic, assign, readwrite) BOOL shouldAnimate;
-
-@end
-
-
 @implementation UIViewController (Testing)
 
 - (void)ar_presentWithFrame:(CGRect)frame
 {
-    SEL animates = NSSelectorFromString(@"setShouldAnimate:");
```
Comment (orta) [inline Artsy_Tests/Extensions/UIViewController+Testing.m]: presuming you did a find all for this and saw we had removed them all?

**Comment by orta** [comment]
Code looks :+1:

File: Artsy_Tests/Extensions/UIViewController+Testing.m (hunk 1 above)
Comment (alloy) [inline Artsy_Tests/Extensions/UIViewController+Testing.m]: Yup

**Comment by alloy** [comment]
> @katarina - have you looked through the videos at how these changes works?

Yeah [she commented in Slack](https://artsy.slack.com/archives/mobile/p1443464838000578):

> the login stuff looks good to me though!

File: Artsy/Networking/API_Modules/ARUserManager.m [hunk 2]
```diff
@@ -663,4 +650,62 @@ - (void)resetTrialUserUUID
...
+{
+    if (ARUserManagerDisableSharedWebCredentials) {
+        NSDictionary *info = @{ NSLocalizedDescriptionKey: @"Developer chose to not use Shared Web Credentials." };
+        completion([NSError errorWithDomain:@"net.artsy.artsy.authentication" code:-1 userInfo:info]);
+        return;
+    }
+
+    SecRequestSharedWebCredential(NULL, NULL, ^(CFArrayRef credentials, CFErrorRef error) {
+        if (error) {
+            // Might just be that there are no credentials available. TODO make that not call the error callback?
```
Comment (alloy) [inline Artsy/Networking/API_Modules/ARUserManager.m]: Eww, that TODO comment can go, will remove it now.

Comment (alloy) [inline Artsy/Networking/API_Modules/ARUserManager.m]: Done

**Comment by orta** [comment]
:+1:

**Pull request merged**